src/DotVVM.Framework/Runtime/Caching/IDotvvmCache.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DotVVM.Framework.Runtime.Caching
+{
+    public interface IDotvvmCache
+    {
+        T GetOrAdd<T>(object key, Func<object, DotvvmCachedItem<T>> factoryFunc);
```

Comment:
If it would have signature `T GetOrAdd<T, TKey>(TKey key, Func<TKey, DotvvmCachedItem<T>> factoryFunc);` there would be no need for casts in the factory.

Reply:
Implemented, I have just changed the order of params to be `<Tkey, TValue>`